accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2671,10 +2686,7 @@ To match a pattern `p` against a value `v`:
 
         1.  Let `r` be the result of `v == c`.
 
-        2.  If `r` is not a Boolean then throw a runtime error. *This can
-            happen if operator `==` on `v`'s type returns `dynamic`.*
-
-        3.  The pattern matches if `r` is false and fails otherwise. *This takes
+        2.  The pattern matches if `r` is false and fails otherwise. *This takes
```

Comment:
I suspect that the implementation teams would complain if we forced them to generate null-check code here that was useless 99% of the time. We already did a similar carve out where we don't do type tests when a pattern's type is a supertype of the matched value type even though that type test would otherwise fail if a legacy `null` flowed in.